<a href="https://colab.research.google.com/github/thaongph/Python-learn/blob/main/first_steps_with_python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First Steps with Python and Jupyter 

![](https://i.imgur.com/gvSnw4A.png)




In [1]:
!git clone https://github.com/thaongph/python-learn.git


Cloning into 'python-learn'...
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 6 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (6/6), 12.38 KiB | 3.09 MiB/s, done.


## Performing Arithmetic Operations using Python

In [ ]:
2 + 3 + 9

14

In [ ]:
99 - 73

26

In [ ]:
23.54 * -1432

-33709.28

In [ ]:
100 / 7

14.285714285714286

In [ ]:
100 // 7

14

In [ ]:
100 % 7

2

In [ ]:
5 ** 3

125

In [ ]:
((2 + 5) * (17 - 3)) / (4 ** 3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 

You can use the empty cells below and add more cells if required.

*There are 86 rooms in a school. 4386 students study there. Equal number of students sits in each room?*

In [1]:
4386/86

51.0

## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [ ]:
500 * (.2 * 1.25)

125.0

In [6]:
profit_per_bag = ice_bag_cost * profit_margin
ice_bag_cost = 1.25
profit_margin = .2
profit_per_bag

0.25

In [8]:
no_of_bag =1000
total_profit= profit_per_bag * no_of_bag
total_profit

250.0

If you try to view the value of a variable that has not been *defined*, i.e., given a value using the assignment statement `variable_name = value`, Python shows an error.

In [9]:
profit

NameError: ignored

Storing and manipulating data using appropriately named variables is a great way to explain what your code does.

Let's display the result of the word problem using a friendly message. We can do this using the `print` *function*.

> **Functions**: A function is a reusable set of instructions. It takes one or more inputs, performs certain operations, and often returns an output. Python provides many in-built functions like `print` and also allows us to define our own functions.

In [11]:
print("Total profit  today is $", total_profit)

Total profit  today is $ 250.0


In [ ]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


Note that we're using the `#` character to add *comments* within our code. 

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines. 

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [22]:
#note for use 
1/1

1.0

In [ ]:
# This comment gets its own line
my_least_favorite_number = 3

In [24]:
example = 5
"""This is a multi-line comment.
Write as little or as much as you'd like.

Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.

Also, if you use good variable names, then your code is
often self explanatory, and you may not even need comments!
"""

example

5



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [25]:
29*200-5000

800

In [26]:
travelers = 29
ticket_price = 200
plane_cost = 5000
profit_of_the_trip= (travelers * ticket_price )- plane_cost
profit_of_the_trip

800

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [31]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [32]:
# Equality check - True
my_favorite_number == 1

True

In [ ]:
 # Equality check - False
my_favorite_number == my_least_favorite_number

False

In [ ]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [ ]:
# Not equal check - False
a_neutral_number != 3

False

In [ ]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [ ]:
# Greater than check - False
my_favorite_number > my_least_favorite_number

False

In [ ]:
# Less than check - True
my_favorite_number < 10

True

In [ ]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [ ]:
# Greater than or equal check - True
my_favorite_number >= 1

True

In [ ]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [ ]:
# Less than or equal check - True
3 + 6 <= 9

True

In [ ]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [ ]:
cost_of_ice_bag = 1.25
is_ice_bag_expensive = cost_of_ice_bag >= 10
print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operator

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |


In [33]:
my_favorite_number

1

In [ ]:
my_favorite_number > 0 and my_favorite_number <= 3

True

In [ ]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [ ]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [ ]:
True and False

False

In [ ]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |



In [ ]:
a_neutral_number = 3

In [ ]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [ ]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [ ]:
my_favorite_number < 0 or True

True

In [ ]:
False or False

False

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [ ]:
not a_neutral_number == 3

False

In [ ]:
not my_favorite_number < 0

True

In [ ]:
not False

True

In [ ]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [ ]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [ ]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [ ]:
not True and 0 < 1 or False and True

False

Experiment with arithmetic, conditional and logical operators in Python using the interactive nature of Jupyter notebooks. We will learn more about variables and functions in future tutorials.

There is also a ternary operator in Python. Learn about it here: https://data-flair.training/blogs/python-ternary-operator/

## Adding text styles using Markdown

`#` for header.

`*` bullet list

`1.` numbered list,

**bold text**

*italic text*

insert [link](https://github.com/thaongph/Python-learn/edit/main/first_steps_with_python.ipynb) to text.

![](https://static.wixstatic.com/media/2cd43b_2373b379948d4e0cb910c593f7edb96e~mv2.png/v1/fill/w_637,h_800,al_c,q_90,enc_auto/2cd43b_2373b379948d4e0cb910c593f7edb96e~mv2.png)

---

a line



```
a paragraph marked
Here is the image of a jaguarr

```

You can learn the full syntax of Markdown here: https://learnxinyminutes.com/docs/markdown/

## Further Reading and References

Following are some resources where you can learn about more arithmetic, conditional and logical operations in Python:

* Python Tutorial at W3Schools: https://www.w3schools.com/python/
* Practical Python Programming: https://dabeaz-course.github.io/practical-python/Notes/Contents.html
* Python official documentation: https://docs.python.org/3/tutorial/index.html

Now that you have taken your first steps with Python, you are ready to move on to the next tutorial: ["A Quick Tour of Variables and Data Types in Python"](https://jovian.ml/aakashns/python-variables-and-data-types).

## Questions for Revision

Try answering the following questions to test your understanding of the topics covered in this notebook:

1. What is a Jupyter notebook? 
2. How do you add a new code cell below an existing cell?
3. How do you add a new Markdown cell below an existing cell?
4. How do you convert a code cell to a Markdown cell or vice versa?
5. How do you execute a code cell within Jupyter?
6. What the different arithmetic operations supported in Python?
7. How do you perform arithmetic operations using Python?
8. What is the difference between the `/` and the `//` operators?
9. What is the difference between the `*` and the `**` operators?
10. What is the order of precedence for arithmetic operators in Python?
11. How do you specify the order in which arithmetic operations are performed in an expression involving multiple operators?
12. How do you solve a multi-step arithmetic word problem using Python?
13. What are variables? Why are they useful?
14. How do you create a variable in Python?
15. What is the assignment operator in Python?
16. What are the rules for naming a variable in Python?
17. How do you view the value of a variable?
18. How do you store the result of an arithmetic expression in a variable?
19. What happens if you try to access a variable that has not been defined?
20. How do you display messages in Python?
21. What type of inputs can the print function accept?
22. What are code comments? How are they useful?
23. What are the different ways of creating comments in Python code?
24. What are the different comparison operations supported in Python?
25. What is the result of a comparison operation?
26. What is the difference between `=` and `==` in Python?
27. What are the logical operators supported in Python?
28. What is the difference between the `and` and `or` operators?
29. Can you use comparison and logical operators in the same expression?
30. What is the purpose of using parentheses in arithmetic or logical expressions?
31. What is Markdown? Why is it useful?
32. How do you create headings of different sizes using Markdown?
33. How do you create bulleted and numbered lists using Markdown?
34. How do you create bold or italic text using Markdown?
35. How do you include links & images within Markdown cells?
36. How do you include code blocks within Markdown cells?
37. Is it possible to execute the code blocks within Markdown cells?
38. How do you upload and share your Jupyter notebook online using Jovian?
39. What is the purpose of the API key requested by jovian.commit ? Where can you find the API key?
40. Where can you learn about arithmetic, conditional and logical operations in Python?
